HTTPS binding support using binding metadata (#3405)
dapr/docs: daprdocs/content/en/reference/components-reference/supported-bindings/http.md

  - name: url
    value: http://something.com
  - name: MTLSRootCA

>     value: /Users/somepath/root.pem # OPTIONAL Secret store ref or <path to root CA> or <pem encoded string>

Reviewer: ```suggestion
    value: /Users/somepath/root.pem # OPTIONAL Secret store ref, <path to root CA>, or <pem encoded string>
```

  - name: MTLSClientCert

>     value: /Users/somepath/client.pem # OPTIONAL Secret store ref or <path to client cert> or <pem encoded string>

Reviewer: ```suggestion
    value: /Users/somepath/client.pem # OPTIONAL Secret store ref, <path to client cert>, or <pem encoded string>
```

  - name: MTLSClientKey

>     value: /Users/somepath/client.key # OPTIONAL Secret store ref or <path to client key> or <pem encoded string>

Reviewer: ```suggestion
    value: /Users/somepath/client.key # OPTIONAL Secret store ref, <path to client key>, or <pem encoded string>
```

| Field              | Required | Binding support | Details | Example |
|--------------------|:--------:|--------|--------|---------|
| url                | Y        | Output |The base URL of the HTTP endpoint to invoke | `http://host:port/path`, `http://myservice:8000/customers`

> | MTLSRootCA         | N        | Output |Secret store reference or Path to root ca certificate or pem encoded string |

Reviewer: ```suggestion
| MTLSRootCA         | N        | Output |Secret store reference, path to root ca certificate, or pem encoded string |
```

> | MTLSClientCert     | N        | Output |Secret store reference or Path to client certificate or pem encoded string  |

Reviewer: ```suggestion
| MTLSClientCert     | N        | Output |Secret store reference, path to client certificate, or pem encoded string  |
```

> | MTLSClientKey      | N        | Output |Secret store reference or Path client private key or pem encoded string |

Reviewer: ```suggestion
| MTLSClientKey      | N        | Output |Secret store reference, path client private key, or pem encoded string |
```

{{< /tabs >}}

{{% alert title="Note" color="primary" %}}

> HTTPS binding support can also be configured using the **MTLSRootCA** metadata option. This will add the specified certificate to the list of trusted certificates for the binding. There is no specific preference for either method. While the **MTLSRootCA** option is easy to use and does not require any changes to the sidecar, it accepts only one certificate. If you need to trust multiple certificates, you need to install them in the sidecar by following the steps above.

Reviewer: ```suggestion
HTTPS binding support can also be configured using the **`MTLSRootCA`** metadata option. This will add the specified certificate to the list of trusted certificates for the binding. There's no specific preference for either method. While the `MTLSRootCA` option is easy to use and doesn't require any changes to the sidecar, it accepts only one certificate. If you need to trust multiple certificates, you need to [install them in the sidecar by following the steps above]({{< ref "#install-the-ssl-certificate-in-the-sidecar" >}}).
```
